battDB/admin.py

```diff
@@ -155,7 +158,32 @@ class DeviceDataInline(common.admin.TabularInline):
     extra = 0
 
 
+class AdminMediaWidget(AdminFileWidget):
+    """
+    Override the default AdminFileWidget to add a download link that uses a SAS token.
+    """
+
+    def render(self, name, value, attrs=None, renderer=None):
+        output = []
+        if value and getattr(value, "url", None):
+            blob_name = value.name
+            blob_url = get_blob_url(value)
+            sas_token = generate_sas_token(blob_name)
+            output.append(f"<a href={blob_url}?{sas_token}>Download File</a>")
+
+        # Put the usual output at the end
+        output.append(super(AdminFileWidget, self).render(name, value, attrs))
+        # Remove the current url from the output that is being replaced
+        if "Currently" in output[-1]:
+            output[-1] = "<br>" + output[-1].split("<br>")[-1]
+
+        return mark_safe("".join(output))
+
+
 class DataFileInline(admin.StackedInline):
+    formfield_overrides = {
+        FileField: {"widget": AdminMediaWidget},
+    }
```

Comment:
I'm guessing this is the trick to override the default widget used in the admin?

Reply:
Yes, exactly. In the end it seemed the path of least resistance.